contracts/src/TokenHome/ERC20TokenHome.sol

```diff
@@ -3,195 +3,18 @@
-    }
-
-    /**
-     * @dev See {TokenHome-_deposit}
-     */
-    function _deposit(uint256 amount) internal virtual override returns (uint256) {
-        ERC20TokenHomeStorage storage $ = _getERC20TokenHomeStorage();
-        return SafeERC20TransferFrom.safeTransferFrom($._token, _msgSender(), amount);
-    }
-
-    /**
-     * @dev See {TokenHome-_withdraw}
-     */
-    function _withdraw(address recipient, uint256 amount) internal virtual override {
-        ERC20TokenHomeStorage storage $ = _getERC20TokenHomeStorage();
-        emit TokensWithdrawn(recipient, amount);
-        $._token.safeTransfer(recipient, amount);
-    }
-
-    /**
-     * @dev See {TokenHome-_handleSendAndCall}
-     *
-     * Approves the recipient contract to spend the amount of tokens from this contract,
-     * and calls {IERC20SendAndCallReceiver-receiveTokens} on the recipient contract.
-     * If the call fails or doesn't spend all of the tokens, the remaining amount is
-     * sent to the fallback recipient.
-     */
-    function _handleSendAndCall(
-        SingleHopCallMessage memory message,
-        uint256 amount
-    ) internal virtual override {
-        ERC20TokenHomeStorage storage $ = _getERC20TokenHomeStorage();
-        IERC20 token = $._token;
-        // Approve the recipient contract to spend the amount from the collateral.
-        SafeERC20.safeIncreaseAllowance($._token, message.recipientContract, amount);
-
-        // Encode the call to {IERC20SendAndCallReceiver-receiveTokens}
-        bytes memory payload = abi.encodeCall(
-            IERC20SendAndCallReceiver.receiveTokens,
-            (
-                message.sourceBlockchainID,
-                message.originTokenTransferrerAddress,
-                message.originSenderAddress,
-                address(token),
-                amount,
-                message.recipientPayload
-            )
-        );
-
-        // Call the recipient contract with the given payload and gas amount.
-        bool success = CallUtils._callWithExactGas(
-            message.recipientGasLimit, message.recipientContract, payload
-        );
-
-        uint256 remainingAllowance = token.allowance(address(this), message.recipientContract);
-
-        // Reset the recipient contract allowance to 0.
-        SafeERC20.forceApprove(token, message.recipientContract, 0);
-
-        if (success) {
-            emit CallSucceeded(message.recipientContract, amount);
-        } else {
-            emit CallFailed(message.recipientContract, amount);
-        }
-
-        // Transfer any remaining allowance to the fallback recipient. This will be the
-        // full amount if the call failed.
-        if (remainingAllowance > 0) {
-            token.safeTransfer(message.fallbackRecipient, remainingAllowance);
-        }
+    ) ERC20TokenHomeUpgradeable(Initializable.Allowed) {
+        initialize(teleporterRegistryAddress, teleporterManager, tokenAddress_, tokenDecimals_);
```

Comment:
Nit: Don't think we need the trailing underscores here anymore since they don't conflict with any variables in this contract itself.

I think this applies in a number of places now since all of the namespaced variable names have leading underscores.

Reply:
updated